src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -552,6 +554,26 @@ private function getArgumentsAsPhp(\DOMElement $node, $name, $file, $lowercase =
                 case 'constant':
                     $arguments[$key] = \constant(trim($arg->nodeValue));
                     break;
+                case 'yaml-file':
+                    if (!\class_exists('Symfony\Component\Yaml\Yaml')) {
+                        throw new \InvalidArgumentException('You need to install the YAML component to parse YAML files.');
+                    }
+
+                    $filePath = $this->container->getParameterBag()->resolveValue((string) $arg->nodeValue);
+                    $rootDir = \dirname($file);
+
+                    if (!\is_file($filePath)) {
+                        $filePath = "$rootDir/$filePath";
+                    }
+
+                    if (!\is_file($filePath)) {
+                        throw new InvalidArgumentException("Unable to locate file \"{$arg->nodeValue}\". Please provide a path relative to \"$rootDir\" or an absolute path.");
+                    }
+
+                    $this->container->addResource(new FileResource($filePath));
+
+                    $arguments[$key] = Yaml::parseFile($filePath, Yaml::PARSE_CONSTANT);
```

Comment:
What about
```php
$arguments[$key] = $this->container->getParameterBag()->resolveValue(
    Yaml::parseFile($filePath, Yaml::PARSE_CONSTANT)
);
```
?
That would allow using nested parameters.

Reply:
Nice idea!

Reply:
not needed. Parameters in arrays are already resolved, and at a better time (late enough that parameter overrides are already applied)